### PR TITLE
Add additional synchronization in Parquet tests

### PR DESCRIPTION
**Pull request by skrzypo987**
This is an attempt to fix flakiness in parquet reader tests. The failures are extremely difficult to reproduce locally. I did manage to produce some sort of quasi-random reproduction in local runs and it does not occur after the fix. I guess we'll see if it works in the CI

## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

fix
> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

tests
> How would you describe this change to a non-technical end user or system administrator?

n/a
## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/ParquetTester.java [hunk 1]
```diff
@@ -601,14 +608,18 @@ public static void writeParquetColumn(
...
-        RecordWriter recordWriter = new TestMapredParquetOutputFormat(parquetSchema, singleLevelArray, dateTimeZone)
-                .getHiveRecordWriter(
-                        jobConf,
-                        new Path(outputFile.toURI()),
-                        Text.class,
-                        compressionCodecName != UNCOMPRESSED,
-                        tableProperties,
-                        () -> {});
+        RecordWriter recordWriter;
+        synchronized (LOCK) {
```
[the remark this answers is not in the document]
Comment (skrzypo987) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/ParquetTester.java:612]: That is a question I don't have an answer to. I would imagine that hive internals are messing with JobConf objects in some magic way.
All my attempts to figure this out led to the conclusion that the bug is impossible, even with multiple threads.
This change simply fixes (maybe?) the flakiness. I guess.

[the remark this answers is not in the document]
Comment (raunaqmorarka) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/ParquetTester.java:612]: TestParquetReader seems to be useful mainly for quicker turn around during development, maybe we just skip that from CI ?

[the remark this answers is not in the document]
Comment (skrzypo987) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/ParquetTester.java:612]: I thought about adding `disabled` property to the `@Test` annotation and a comment to remove it if you want to run manually

[the remark this answers is not in the document]
Comment (raunaqmorarka) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/ParquetTester.java:612]: That's fine too

**Comment by skrzypo987** [comment]
Closing in favor of #13102

**Pull request closed without merging**